### PR TITLE
Rename PayID Code Idiomatically

### PR DESCRIPTION
**Pull request by keefertaylor**
## High Level Overview of Change

JavaScript prefers to use `PayId` to `PayID`. This is a large one shot PR to fix that. 

### Context of Change

This is a large in place refactor. We rely on our tests to keep us honest that functionality has not changed. 

### Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Refactor (non-breaking change that only restructures code)
- [ ] Tests (You added tests for code that already exists, or your new feature included in this PR)
- [ ] Documentation Updates
- [ ] Release

## Before / After

Closer to idiomatic and consistent naming. 

## Test Plan

CI

### CODE REVIEW

File: src/PayID/pay-id-client.ts [hunk 1]
```diff
@@ -43,6 +52,140 @@ export default class PayIDClient {
...
+    payID: string,
+    nonce: string,
+  ): Promise<SignatureWrapperInvoice> {
+    return this.wrappedPayIdClient.getInvoice(payID, nonce)
+  }
+
+  /**
+   * Post an invoice with compliance data.
+   *
+   * @param payID The Pay ID compliance data is associated with.
```
Comment (LoisRP) [inline src/PayID/pay-id-client.ts:85]: Pay ID ->PayID

File: src/PayID/pay-id-client.ts [hunk 2]
```diff
@@ -43,6 +52,140 @@ export default class PayIDClient {
...
+   */
+  async getInvoice(
+    payID: string,
+    nonce: string,
+  ): Promise<SignatureWrapperInvoice> {
+    return this.wrappedPayIdClient.getInvoice(payID, nonce)
+  }
+
+  /**
+   * Post an invoice with compliance data.
```
Comment (LoisRP) [inline src/PayID/pay-id-client.ts:83]: Are we still using invoice now?

File: src/PayID/pay-id-client.ts [hunk 3]
```diff
@@ -43,6 +52,140 @@ export default class PayIDClient {
...
+   * @returns An address representing the given PayID.
+   */
+  async addressForPayID(payID: string): Promise<CryptoAddressDetails> {
+    return this.wrappedPayIdClient.addressForPayId(payID)
+  }
+
+  /**
+   * Generate a new invoice with compliance requests.
+   *
+   * @param payID The Pay ID to request an invoice for.
```
Comment (LoisRP) [inline src/PayID/pay-id-client.ts:72]: Pay ID -> PayID (throughout)

File: src/PayID/pay-id-client.ts [hunk 4]
```diff
@@ -43,6 +52,140 @@ export default class PayIDClient {
...
+   * @param nonce A randomly selected nonce that is unique to this invoice request.
```
Comment (LoisRP) [inline src/PayID/pay-id-client.ts:73]: For some reason, I thought nonce was no longer going to be used? I realize that's outside the scope of this review.

File: src/PayID/pay-id-client.ts [hunk 5]
```diff
@@ -43,6 +52,140 @@ export default class PayIDClient {
...
+   * @param originatorUserLegalName The legal name of the originator.
+   * @param originatorAccountID The account ID of the originator.
+   * @param originatorUserPhysicalAddress The physical address of the originator.
+   * @param originatorInstitutionName The institution name of the originator.
+   * @param valueAmount The value being transferred.
+   * @param valueScale The scale of the value.
+   * @param timestamp The time that the operation occurred.
+   * @param beneficiaryInstitutionName The beneficiary insitution name.
+   * @param beneficiaryUserLegalName The legal name of the receiver at the beneficiary institution. Optional, defaults to undefined.
+   * @param beneficiaryUserPhysicalAddress The phsyical address of the receiver at the beneficiary institution. Optional, defaults to undefined.
```
Comment (LoisRP) [inline src/PayID/pay-id-client.ts:99]: physical - typo

File: src/PayID/pay-id-client.ts [hunk 6]
```diff
@@ -43,6 +52,140 @@ export default class PayIDClient {
...
+   * @param publicKeyType The type of public key.
+   * @param publicKeyData An array of public keys which lead back to the root trust certificate.
+   * @param publicKey The public key.
+   * @param signature The signature of the operation.
+   * @param originatorUserLegalName The legal name of the originator.
+   * @param originatorAccountID The account ID of the originator.
+   * @param originatorUserPhysicalAddress The physical address of the originator.
+   * @param originatorInstitutionName The institution name of the originator.
+   * @param valueAmount The value being transferred.
+   * @param valueScale The scale of the value.
```
Comment (LoisRP) [inline src/PayID/pay-id-client.ts:95]: The scale of the value. - Is it clear what this is, or what the value means?

Comment (keefertaylor) [reply src/PayID/pay-id-client.ts:95]: Ah, I think I should say `valueAmount`? 

Happy to hear other revisions to this documentation to. 

The idea is that $1 USD at scale 2 is 100. Or $1.23USD is 123 at scale 2. Scale is the number of significant decimal places? 

This is terminology that is borrowed from ILP. All their APIs take a value and a scale. Maybe they've found a better way to describe it?

File: src/PayID/pay-id-client.ts (hunk 4 above)
Comment (keefertaylor) [reply src/PayID/pay-id-client.ts:73]: Same comment here. Queried the PayID folks.

File: src/PayID/pay-id-client.ts (hunk 2 above)
Comment (keefertaylor) [reply src/PayID/pay-id-client.ts:83]: AFAIK we are but I could be out of the loop. 

I queried the PayID folks and I'll leave this PR open until I get a definitive answer.

Comment (hbergren) [reply src/PayID/pay-id-client.ts:83]: This is `paymentSetupDetails` now... but Travel Rule is pretty faked and not real, so do what you will.

File: src/PayID/pay-id-client.ts (hunk 4 above)
Comment (hbergren) [reply src/PayID/pay-id-client.ts:73]: Yeah, we're not using `nonce` anymore. I think the field we use now is `txId`, which gets provided by the PayID server on a `GET /payment-setup-details`, and which the client should provide for a `POST /payment-setup-details` and `POST /payment-proof`.

File: src/PayID/pay-id-client.ts (hunk 6 above)
Comment (hbergren) [reply src/PayID/pay-id-client.ts:95]: I think XRPL uses the same notion of an amount and scale.

**Pull request merged**